lib/unleash_proxy_client_flutter.dart

```diff
@@ -355,7 +353,12 @@ class UnleashClient extends EventEmitter {
 
   Future<void> setContextFields(Map<String, String> fields) async {
```

Comment:
i copied all 3 await conditions from updateContext to setContextFields. This way we have same logic for all startup corner cases.